util.go

```diff
@@ -13,10 +13,22 @@ func RetrieveLastBlockFromStorage(s Storage) (Block, error) {
 	if height == 0 {
 		return nil, nil
 	}
-
 	lastBlock, _, retrieved := s.Retrieve(height - 1)
 	if !retrieved {
 		return nil, fmt.Errorf("failed retrieving last block from storage with seq %d", height-1)
 	}
 	return lastBlock, nil
 }
+
+func RetrieveLastFromStorage(s Storage) (Block, FinalizationCertificate, error) {
```

Comment:
There is not much use in making a new method almost identical as the one above. 

We can just change the function above to return an fCert and just use it. 

I'd say alternatively delete the method above and use this one, but the above method has a unit test and this doesn't.